### ISSUE TITLE
Zero shot - evaluate labels independently

### ISSUE THREAD

**Issue by subhamkhemka** [openai/CLIP#224]
Hi

How can I consider the labels to be independent of each other ? Do not want the probs summing up to 1 instead want to evaluate each label on its own, sort of a multi label classification.

```
import torch
import clip
from PIL import Image

device = "cuda" if torch.cuda.is_available() else "cpu"
model, preprocess = clip.load("ViT-B/32", device=device)

image = preprocess(Image.open("CLIP.png")).unsqueeze(0).to(device)
text = clip.tokenize(["a diagram", "a dog", "a cat"]).to(device)

with torch.no_grad():
    image_features = model.encode_image(image)
    text_features = model.encode_text(text)
    
    logits_per_image, logits_per_text = model(image, text)
    probs = logits_per_image.softmax(dim=-1).cpu().numpy()

print("Label probs:", probs)  # prints: [[0.9927937  0.00421068 0.00299572]]
```

Thanks

**Comment by justlike-prog** [comment]
@jongwook @subhamkhemka So there is no way to get a per word probability rather than just the similarity? It seems to be quite hard to find a good threshold just by similarity.
